src/Octoshift/Services/OctoLogger.cs

```diff
@@ -157,4 +164,9 @@ public virtual void LogSuccess(string msg)
     }
 
     public virtual void RegisterSecret(string secret) => _secrets.Add(secret);
+
+    public virtual void AddRedactionPattern(Regex pattern)
+    {
+        _redactionPatterns.Add(pattern);
+    }
```

Comment:
Instead of exposing a public method to register regex patters, I would prefer to keep a list of all patterns internally.